File: examples/go/escrow/Escrow_test.go
package contract

import (
	"testing"
	runar "github.com/icellan/runar/packages/runar-go"
)

func newEscrow() *Escrow {
	return &Escrow{
		Buyer:   runar.Alice.PubKey,
		Seller:  runar.Bob.PubKey,
		Arbiter: runar.Charlie.PubKey,
	}
}

func TestEscrow_Release(t *testing.T) {
	sellerSig := runar.SignTestMessage(runar.Bob.PrivKey)
	arbiterSig := runar.SignTestMessage(runar.Charlie.PrivKey)
	newEscrow().Release(sellerSig, arbiterSig)
}

func TestEscrow_Refund(t *testing.T) {
	buyerSig := runar.SignTestMessage(runar.Alice.PrivKey)
	arbiterSig := runar.SignTestMessage(runar.Charlie.PrivKey)
	newEscrow().Refund(buyerSig, arbiterSig)
}

func TestEscrow_Compile(t *testing.T) {
	if err := runar.CompileCheck("Escrow.runar.go"); err != nil {
		t.Fatalf("Rúnar compile check failed: %v", err)
	}
}

File: examples/rust/escrow/Escrow_test.rs
#[path = "Escrow.runar.rs"]
mod contract;

use contract::*;
use runar::prelude::*;

fn new_escrow() -> Escrow {
    Escrow {
        buyer: ALICE.pub_key.to_vec(),
        seller: BOB.pub_key.to_vec(),
        arbiter: CHARLIE.pub_key.to_vec(),
    }
}

#[test]
fn test_release() {
    new_escrow().release(&BOB.sign_test_message(), &CHARLIE.sign_test_message());
}

#[test]
fn test_refund() {
    new_escrow().refund(&ALICE.sign_test_message(), &CHARLIE.sign_test_message());
}

#[test]
fn test_compile() {
    runar::compile_check(include_str!("Escrow.runar.rs"), "Escrow.runar.rs").unwrap();
}
